api/app/graphql/mutations/sign_out.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module Mutations
+  class SignOut < BaseMutation
+    include JsonWebToken
+
+    graphql_name 'SignOut'
+
+    argument :token, String, required: true
+
+    field :token, String, null: true
+
+    def resolve(args)
+      decoded_token = decode_token(args[:token])
+      raise GraphQL::ExecutionError, 'Invalid token' unless decoded_token.present?
+
+      token_id = decoded_token[0]['id']
+      raise GraphQL::ExecutionError, 'Invalid token' unless token_id.present?
```

Comment:
Is there a cleaner way of raising these execution errors?

Reply:
I think this depends on the flow here.

Answering the error question:
I would structure this by having decode_token take in the 'token' and then expose an object that responds to `.id` so that you only have one call here.  

The decode_token method would raise it's own exception that you can handle at the graphql_controller level so that anywhere you do this it will ensure everything is good.

That said, I would move the logout completely out of graphql so that you have a more standard interaction here.

Reply:
I fixed this by refactoring the token decoding logic into a class method `find_by_jwt` on `AuthToken`. I think that makes this behavior much more reusable when on the model since we need that in a couple places.